### PR TITLE
feat: stubs vagrant docs and tutorials routes

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-onboard-vagrant-hashicorp.vercel.app/vagrant) 🔎
- [Asana task](https://app.asana.com/0/1202702246158644/1202614566185926) 🎟️

## 🗒️ What

This PR stubs out the vagrant docs and tutorial routes. The landing page content is FPO, similar to the nav items and top level sidebar items. I presume those will be adjusted when design takes a look at vagrant. 

## 🧪 Testing

- Visit [/vagrant](https://dev-portal-git-onboard-vagrant-hashicorp.vercel.app/vagrant)
- [/vagrant/downloads](https://dev-portal-git-onboard-vagrant-hashicorp.vercel.app/vagrant/downloads)
- [/vagrant/docs](https://dev-portal-git-onboard-vagrant-hashicorp.vercel.app/vagrant/docs)
- [/vagrant/docs/doc](https://dev-portal-git-onboard-vagrant-hashicorp.vercel.app/vagrant/docs/installation)
- [/vagrant/tutorials](https://dev-portal-git-onboard-vagrant-hashicorp.vercel.app/vagrant/tutorials)
- [/vagrant/tutorials/collection](https://dev-portal-git-onboard-vagrant-hashicorp.vercel.app/vagrant/tutorials/getting-started)
- [/vagrant/tutorials/tutorial](https://dev-portal-git-onboard-vagrant-hashicorp.vercel.app/vagrant/tutorials/getting-started/getting-started-index)

## 💭 Anything else?
**Related follow up tasks** 
- [follow up on vmware downloads ticket](https://app.asana.com/0/1202097197789424/1201945191425824)
- [follow up on vagrant-cloud route structure ticket](https://app.asana.com/0/1202520168081556/1202764105038791)
- there are some duplicate 'overview' items in the sidebar content. We'll handle that during [design refinement](https://app.asana.com/0/1202520168081556/1202580683124096)
Screen Shot 2022-08-08 at 3 05 33 PM

- The icons for the docs nav items are all the same, will address during [design refinement.](https://app.asana.com/0/1202520168081556/1202580683124096)
Screen Shot 2022-08-08 at 3 05 36 PM

### CODE REVIEW

File: src/content/vagrant/docs-landing.json
Comment (kendallstrautman) [inline src/content/vagrant/docs-landing.json:1]: Copid this from consul, TODO - figure out what vagrant needs.

File: src/content/vagrant/docs-landing.json
Comment (kendallstrautman) [inline src/content/vagrant/docs-landing.json:2]: I copy / pasta'd this from the other stubbed landing views 🍝

File: src/data/vagrant.json
Comment (kendallstrautman) [inline src/data/vagrant.json:79]: These icons are all the same. Not sure if all these items should be in here also, esp 'introduction'?

File: src/lib/remark-plugins/rewrite-tutorial-links/__tests__/rewrite-tutorial-links.test.ts
Comment (kendallstrautman) [inline src/lib/remark-plugins/rewrite-tutorial-links/__tests__/rewrite-tutorial-links.test.ts:10]: Copied this from zachs #774, if that gets merged first i'll rebase.

File: src/data/vagrant.json
Comment (ashleemboyer) [inline src/data/vagrant.json:108]: For the sidebar on /vagrant, we need to list this General Documentation item first.

File: src/data/vagrant.json
Comment (ashleemboyer) [reply src/data/vagrant.json:79]: Icons are good for now! 😊 And we do want to include all of the `rootDocsPaths` here. So you're good to go! ✨

File: src/data/vagrant.json
Comment (ashleemboyer) [inline src/data/vagrant.json:93]: Since [/vagrant/intro](https://dev-portal-git-onboard-vagrant-hashicorp.vercel.app/vagrant/intro) and [/vagrant/vagrant-cloud](https://dev-portal-git-onboard-vagrant-hashicorp.vercel.app/vagrant/vagrant-cloud) are the current URLs for those sections of docs, let's set those as the `pathSuffix` values here for them.

```suggestion
			{
				"icon": "docs",
				"label": "Introduction",
				"pathSuffix": "intro"
			},
			{
				"icon": "docs",
				"label": "General Documentation",
				"pathSuffix": "docs"
			},
			{
				"icon": "docs",
				"label": "Vagrant Cloud",
				"pathSuffix": "vagrant-cloud"
			}
```

**Review by ashleemboyer — CHANGES_REQUESTED** [review]
Overall looks great! 🙌🏻  Left a couple of small comments to resolve before merging.

**Review by zchsh — APPROVED** [review]
✅ Looks good to me!

+1 to Ashlee's suggestion above to fix up `pathSuffix` in `navigationHeaderItems`, as otherwise the dropdown in the nav doesn't work. Approving assuming that gets applied, though since this is a stub also kinda fine with merging as-is 👍

Side note: I really dig 🎟️ [[NavigationHeader] refactor Documentation dropdown to use rootDocsPaths](https://app.asana.com/0/1202097197789424/1202765914032417/f) as a way to improve the DX for onboarding routes, since having to manually keep `rootDocsPath` and `navigationHeaderItems` in sync when they're the same does feel kinda meh. Thanks for making that ticket Ashlee 🙌

File: src/data/vagrant.json
Comment (kendallstrautman) [reply src/data/vagrant.json:108]: Cool, will update!

Comment (kendallstrautman) [reply src/data/vagrant.json]: I just realized that `rootDocsPaths` was already defined in the data file 😶‍🌫️ lol so I removed it from this PR!

**Review by ashleemboyer — APPROVED** [review]
👏🏻

**Pull request merged**
